tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -212,6 +214,32 @@ def sync(sync_ttl, node_status_ttl):
                     detected_cluster_tag = "detected_cluster/%s" % \
                                            sds_details[
                                                'detected_cluster_id']
+                    # Detected cluster id will change when new node
+                    # added into peer list and when peer detach happens,
+                    # Node_context should not maintain multiple DC ids
+                    old_dc_id = "detected_cluster/%s" % dc.detected_cluster_id
```

Comment:
I see that `old_dc_id` you get from `detected_cluster/{int_id}` and at line#214 `detected_cluster_tag` also as `detected_cluster/{sds_detail's detected_cluster_id}`? Do you mean if nodes removed / added from CLI the value of `sds_details['detected_cluster_id']` would differ and so no match between `old_dc_id` and `detected_cluster_tag`?

Reply:
yes, an old-detected cluster is no longer when we added a new node.  So sds_details['detected_cluster_id'] should be replaced by old-id. No need to maintain both the detected cluster_id ETCD.